net/src/client/http/mod.rs

```diff
@@ -0,0 +1,351 @@
+        let version = req
+            .version()
+            .ok_or(WitnetHttpError::UnknownVersion {
+                version: String::from("None"),
+            })
+            .and_then(WitnetHttpVersion::try_from)
+            .map(isahc::http::Version::from)
+            .unwrap_or_default();
+        let uri = req.url().to_string();
+        let body = isahc::AsyncBody::from_reader(req.take_body().into_reader());
+        let headers: Vec<(String, String)> = req
+            .header_names()
+            .map(|name| {
+                (
+                    name.to_string(),
+                    req.header(name)
+                        .map(std::string::ToString::to_string)
+                        .unwrap_or_default(),
+                )
+            })
+            .collect();
+
+        // Start to build an isahc request with the basic parts
+        let mut req = isahc::http::Request::builder()
+            .method(method)
+            .version(version)
+            .uri(uri);
+
+        // Attach the headers to the builder
+        for (key, value) in headers {
+            req = req.header(key, value)
+        }
+
+        // Attach the body to the builder and compose the request itself
+        let req = req
+            .body(body)
+            .map_err(|err| Self::Error::HttpRequestError {
+                msg: err.to_string(),
+            })?;
+
+        Ok(WitnetHttpRequest::from(req))
+    }
+}
+
+/// Enables interoperability between `isahc::Response` and `surf::http::Response`.
+pub struct WitnetHttpResponse {
+    res: isahc::Response<isahc::AsyncBody>,
+}
+
+impl From<isahc::Response<isahc::AsyncBody>> for WitnetHttpResponse {
+    #[inline]
+    fn from(res: isahc::Response<isahc::AsyncBody>) -> Self {
+        Self { res }
+    }
+}
+
+impl TryFrom<WitnetHttpResponse> for surf::http::Response {
+    type Error = WitnetHttpError;
+
+    fn try_from(res: WitnetHttpResponse) -> Result<Self, Self::Error> {
+        // Get the different parts of the isahc response
+        let (parts, body) = res.res.into_parts();
+        let status = WitnetHttpStatusCode::from(parts.status);
+        let version = WitnetHttpVersion::from(parts.version);
+        let headers: Vec<(String, String)> = parts
+            .headers
+            .iter()
+            .map(|(key, value)| {
+                (
+                    key.to_string(),
+                    value.to_str().unwrap_or_default().to_string(),
+                )
+            })
+            .collect();
+        let body_reader = futures::io::BufReader::new(body);
+
+        // Create a surf response and set all the relevant parts
+        let mut res = surf::http::Response::new(status);
+        res.set_version(Some(surf::http::Version::try_from(version)?));
+        res.set_body(surf::Body::from_reader(body_reader, None));
+        for (key, value) in headers {
+            res.insert_header(key.as_str(), value);
```

Comment:
I believe this is not needed because we never check response headers, and there is no way to do that in RADON, but we can leave it for future compatibility.

Reply:
Same applies to `res.set_version`.